Six new winter weather diags

ericaligo-NOAA opened the pull request:
Six new winter weather diagnostics from GFS and RRFS are read in. Changes were made to SURFCE.f to allow the correct time stamp to be used for the continuous fields. The new fields were also added to the xml files and new flat files were created.

Issue: 
https://github.com/NOAA-EMC/UPP/issues/568

ericaligo-NOAA committed:
read in six new winter weather diags, allocate, deallocate,and add new vrbs (3 of them), plus for continuous fields modify timestamp for FV3R and GFS and add fields to the GFS and fv3lam xml and flat files.

ericaligo-NOAA committed:
Rename winter wx diags

HuiyaChuang-NOAA commented on sorc/ncep_post.fd/INITPOST_NETCDF.f:
```diff
@@ -2260,6 +2261,35 @@ SUBROUTINE INITPOST_NETCDF(ncid2d,ncid3d)
       call read_netcdf_2d_para(ncid2d,ista,ista_2l,iend,iend_2u,jsta,jsta_2l,jend,jend_2u, &
       spval,VarName,SFCEXC)
 
+! accumulated snowfall
```
@ericaligo-NOAA hoping you can clarify for me.
UPP already reads in "snod" from FV3 model output which is supposed to be snow depth.  
What is the difference between snod and acsnow?
Also, I believe you said "acsnow" is water equivalent of snow fall, is that correct?

ericaligo-NOAA commented:
Hi,

ACSNOW is the water equivalent of snowfall from the microphysics. I 
updated codes and changed the name of the fields read into the UPP.  I 
renamed ACSNOW to TSNOWP.

Eric

On 12/12/2022 3:46 PM, HuiyaChuang-NOAA wrote:
>
> ***@***.**** commented on this pull request.
>
> ------------------------------------------------------------------------
>
> In sorc/ncep_post.fd/INITPOST_NETCDF.f 
> <https://github.com/NOAA-EMC/UPP/pull/600#discussion_r1046373678>:
>
> > @@ -2260,6 +2261,35 @@ SUBROUTINE INITPOST_NETCDF(ncid2d,ncid3d)
>         call read_netcdf_2d_para(ncid2d,ista,ista_2l,iend,iend_2u,jsta,jsta_2l,jend,jend_2u, &
>         spval,VarName,SFCEXC)
>   
> +! accumulated snowfall
>
> @ericaligo-NOAA <https://github.com/ericaligo-NOAA> hoping you can 
> clarify for me.
> UPP already reads in "snod" from FV3 model output which is supposed to 
> be snow depth.
> What is the difference between snod and acsnow?
> Also, I believe you said "acsnow" is water equivalent of snow fall, is 
> that correct?
>
> —
> Reply to this email directly, view it on GitHub 
> <https://github.com/NOAA-EMC/UPP/pull/600#pullrequestreview-1214316375>, 
> or unsubscribe 
> <https://github.com/notifications/unsubscribe-auth/ALQ75MIEBZAAWNYVECIKVWDWM6FKJANCNFSM6AAAAAASYSFTNM>.
> You are receiving this because you were mentioned.Message ID: 
> ***@***.***>
>

WenMeng-NOAA commented on parm/POST_XML_Flat_File_processor_POST_FlatFile.log.20221208:
```diff
@@ -0,0 +1,144 @@
+173403: [00138]  Begin PostXMLPreprocessor 
```
@ericaligo-NOAA This log file is not needed to be committed. Please remove it.

ericaligo-NOAA committed:
removed log file

ericaligo-NOAA commented:
Done.

On 12/18/2022 6:11 PM, WenMeng-NOAA wrote:
>
> ***@***.**** commented on this pull request.
>
> ------------------------------------------------------------------------
>
> In parm/POST_XML_Flat_File_processor_POST_FlatFile.log.20221208 
> <https://github.com/NOAA-EMC/UPP/pull/600#discussion_r1051682954>:
>
> > @@ -0,0 +1,144 @@
> +173403: [00138]  Begin PostXMLPreprocessor
>
> @ericaligo-NOAA <https://github.com/ericaligo-NOAA> This log file is 
> not needed to be committed. Please remove it.
>
> —
> Reply to this email directly, view it on GitHub 
> <https://github.com/NOAA-EMC/UPP/pull/600#pullrequestreview-1222107844>, 
> or unsubscribe 
> <https://github.com/notifications/unsubscribe-auth/ALQ75MNWF3LQHTFSPCT4C5LWN6K2DANCNFSM6AAAAAASYSFTNM>.
> You are receiving this because you were mentioned.Message ID: 
> ***@***.***>
>

WenMeng-NOAA commented on parm/postcntrl_gfs_f00.xml:
```diff
@@ -907,6 +907,36 @@
        <scale>3.0</scale>
     </param>
 
+    <param>
```
@ericaligo-NOAA Do you think these six winter diag. variables will be calculated in model at F00? If not, you don't need adding them in the control files with f00.

ericaligo-NOAA commented:
Yes, they will be calculated in the model. Do you think the three 
continuous fields need to be in F00 for restart runs?

On 12/19/2022 9:02 AM, WenMeng-NOAA wrote:
>
> ***@***.**** commented on this pull request.
>
> ------------------------------------------------------------------------
>
> In parm/postcntrl_gfs_f00.xml 
> <https://github.com/NOAA-EMC/UPP/pull/600#discussion_r1052231965>:
>
> > @@ -907,6 +907,36 @@
>          <scale>3.0</scale>
>       </param>
>   
> +    <param>
>
> @ericaligo-NOAA <https://github.com/ericaligo-NOAA> Do you think these 
> six winter diag. variables will be calculated in model at F00? If not, 
> you don't need adding them in the control files with f00.
>
> —
> Reply to this email directly, view it on GitHub 
> <https://github.com/NOAA-EMC/UPP/pull/600#pullrequestreview-1222950371>, 
> or unsubscribe 
> <https://github.com/notifications/unsubscribe-auth/ALQ75MNY5ZAMD2STIXLG2ALWOBTIRANCNFSM6AAAAAASYSFTNM>.
> You are receiving this because you were mentioned.Message ID: 
> ***@***.***>
>

WenMeng-NOAA commented:
> Yes, they will be calculated in the model. Do you think the three continuous fields need to be in F00 for restart runs?
> […](#)
> On 12/19/2022 9:02 AM, WenMeng-NOAA wrote: ***@***.**** commented on this pull request. ------------------------------------------------------------------------ In parm/postcntrl_gfs_f00.xml <[#600 (comment)](https://github.com/NOAA-EMC/UPP/pull/600#discussion_r1052231965)>: > @@ -907,6 +907,36 @@ <scale>3.0</scale> </param> + <param> @ericaligo-NOAA <https://github.com/ericaligo-NOAA> Do you think these six winter diag. variables will be calculated in model at F00? If not, you don't need adding them in the control files with f00. — Reply to this email directly, view it on GitHub <[#600 (review)](https://github.com/NOAA-EMC/UPP/pull/600#pullrequestreview-1222950371)>, or unsubscribe <https://github.com/notifications/unsubscribe-auth/ALQ75MNY5ZAMD2STIXLG2ALWOBTIRANCNFSM6AAAAAASYSFTNM>. You are receiving this because you were mentioned.Message ID: ***@***.***>

@ericaligo-NOAA Are you meaning the restart run for model side or UPP side?

ericaligo-NOAA commented:
Restart for the model. If this has nothing to do with the 
parm/postcntrl_gfs_f00.xml 
<https://github.com/NOAA-EMC/UPP/pull/600#discussion_r1052231965> then I 
can remove the six fields from that file.

On 12/19/2022 9:23 AM, WenMeng-NOAA wrote:
>
>     Yes, they will be calculated in the model. Do you think the three
>     continuous fields need to be in F00 for restart runs?
>     … <#>
>     On 12/19/2022 9:02 AM, WenMeng-NOAA wrote: /*@*/.**** commented on
>     this pull request.
>     ------------------------------------------------------------------------
>     In parm/postcntrl_gfs_f00.xml <#600 (comment)
>     <https://github.com/NOAA-EMC/UPP/pull/600#discussion_r1052231965>>:
>     > @@ -907,6 +907,36 @@ 3.0 + @ericaligo-NOAA
>     <https://github.com/ericaligo-NOAA>
>     https://github.com/ericaligo-NOAA Do you think these six winter
>     diag. variables will be calculated in model at F00? If not, you
>     don't need adding them in the control files with f00. — Reply to
>     this email directly, view it on GitHub <#600 (review)
>     <https://github.com/NOAA-EMC/UPP/pull/600#pullrequestreview-1222950371>>,
>     or unsubscribe
>     https://github.com/notifications/unsubscribe-auth/ALQ75MNY5ZAMD2STIXLG2ALWOBTIRANCNFSM6AAAAAASYSFTNM.
>     You are receiving this because you were mentioned.Message ID:
>     /*@*/.***>
>
> @ericaligo-NOAA <https://github.com/ericaligo-NOAA> Are you meaning 
> the restart run for model side or UPP side?
>
> —
> Reply to this email directly, view it on GitHub 
> <https://github.com/NOAA-EMC/UPP/pull/600#issuecomment-1357748847>, or 
> unsubscribe 
> <https://github.com/notifications/unsubscribe-auth/ALQ75MJAX4RP27QEZRFTYCLWOBVVNANCNFSM6AAAAAASYSFTNM>.
> You are receiving this because you were mentioned.Message ID: 
> ***@***.***>
>

WenMeng-NOAA commented:
> Restart for the model. If this has nothing to do with the parm/postcntrl_gfs_f00.xml <[#600 (comment)](https://github.com/NOAA-EMC/UPP/pull/600#discussion_r1052231965)> then I can remove the six fields from that file.
> […](#)
> On 12/19/2022 9:23 AM, WenMeng-NOAA wrote: Yes, they will be calculated in the model. Do you think the three continuous fields need to be in F00 for restart runs? … <#> On 12/19/2022 9:02 AM, WenMeng-NOAA wrote: /*@*/.**** commented on this pull request. ------------------------------------------------------------------------ In parm/postcntrl_gfs_f00.xml <#600 (comment) <[#600 (comment)](https://github.com/NOAA-EMC/UPP/pull/600#discussion_r1052231965)>>: > @@ -907,6 +907,36 @@ 3.0 + @ericaligo-NOAA <https://github.com/ericaligo-NOAA> https://github.com/ericaligo-NOAA Do you think these six winter diag. variables will be calculated in model at F00? If not, you don't need adding them in the control files with f00. — Reply to this email directly, view it on GitHub <#600 (review) <[#600 (review)](https://github.com/NOAA-EMC/UPP/pull/600#pullrequestreview-1222950371)>>, or unsubscribe https://github.com/notifications/unsubscribe-auth/ALQ75MNY5ZAMD2STIXLG2ALWOBTIRANCNFSM6AAAAAASYSFTNM. You are receiving this because you were mentioned.Message ID: /*@*/.***> @ericaligo-NOAA <https://github.com/ericaligo-NOAA> Are you meaning the restart run for model side or UPP side? — Reply to this email directly, view it on GitHub <[#600 (comment)](https://github.com/NOAA-EMC/UPP/pull/600#issuecomment-1357748847)>, or unsubscribe <https://github.com/notifications/unsubscribe-auth/ALQ75MJAX4RP27QEZRFTYCLWOBVVNANCNFSM6AAAAAASYSFTNM>. You are receiving this because you were mentioned.Message ID: ***@***.***>

@ericaligo-NOAA I don't think the UPP grib2 file are needed for GFS restart run. You might confirm with GFS developers.

ericaligo-NOAA committed:
remove six winter wx diags from f00 file

ericaligo-NOAA commented:
Model restarts are unrelated to grib2 files according to GFS experts. I 
removed the six winter weather diags from parm/postcntrl_gfs_f00.xml

On 12/19/2022 9:37 AM, WenMeng-NOAA wrote:
>
>     Restart for the model. If this has nothing to do with the
>     parm/postcntrl_gfs_f00.xml <#600 (comment)
>     <https://github.com/NOAA-EMC/UPP/pull/600#discussion_r1052231965>>
>     then I can remove the six fields from that file.
>     … <#>
>     On 12/19/2022 9:23 AM, WenMeng-NOAA wrote: Yes, they will be
>     calculated in the model. Do you think the three continuous fields
>     need to be in F00 for restart runs? … <#> On 12/19/2022 9:02 AM,
>     WenMeng-NOAA wrote: //@//.**** commented on this pull request.
>     ------------------------------------------------------------------------
>     In parm/postcntrl_gfs_f00.xml <#600
>     <https://github.com/NOAA-EMC/UPP/pull/600> (comment) <#600
>     (comment)
>     <https://github.com/NOAA-EMC/UPP/pull/600#discussion_r1052231965>>>:
>     > @@ -907,6 +907,36 @@ 3.0 + @ericaligo-NOAA
>     <https://github.com/ericaligo-NOAA>
>     https://github.com/ericaligo-NOAA
>     https://github.com/ericaligo-NOAA Do you think these six winter
>     diag. variables will be calculated in model at F00? If not, you
>     don't need adding them in the control files with f00. — Reply to
>     this email directly, view it on GitHub <#600
>     <https://github.com/NOAA-EMC/UPP/pull/600> (review) <#600 (review)
>     <https://github.com/NOAA-EMC/UPP/pull/600#pullrequestreview-1222950371>>>,
>     or unsubscribe
>     https://github.com/notifications/unsubscribe-auth/ALQ75MNY5ZAMD2STIXLG2ALWOBTIRANCNFSM6AAAAAASYSFTNM.
>     You are receiving this because you were mentioned.Message ID:
>     //@//./*> @ericaligo-NOAA <https://github.com/ericaligo-NOAA>
>     https://github.com/ericaligo-NOAA Are you meaning the restart run
>     for model side or UPP side? — Reply to this email directly, view
>     it on GitHub <#600 (comment)
>     <https://github.com/NOAA-EMC/UPP/pull/600#issuecomment-1357748847>>,
>     or unsubscribe
>     https://github.com/notifications/unsubscribe-auth/ALQ75MJAX4RP27QEZRFTYCLWOBVVNANCNFSM6AAAAAASYSFTNM.
>     You are receiving this because you were mentioned.Message ID:
>     /*@*/.*/>
>
> @ericaligo-NOAA <https://github.com/ericaligo-NOAA> I don't think the 
> UPP grib2 file are needed for GFS restart run. You might confirm with 
> GFS developers.
>
> —
> Reply to this email directly, view it on GitHub 
> <https://github.com/NOAA-EMC/UPP/pull/600#issuecomment-1357765498>, or 
> unsubscribe 
> <https://github.com/notifications/unsubscribe-auth/ALQ75MKLIEGILWGHXGHJLQ3WOBXKJANCNFSM6AAAAAASYSFTNM>.
> You are receiving this because you were mentioned.Message ID: 
> ***@***.***>
>

WenMeng-NOAA commented:
@ericaligo-NOAA Can you sync your branch with upstream/develop? The procedure can be found at
https://github.com/NOAA-EMC/UPP/wiki/UPP-Code-Development#update-your-branch-at-your-fork-with-branch-develop-at-authoritative-repository

ericaligo-NOAA commented:
I tried but I get this error message:

Hera:/scratch1/NCEPDEV/stmp2/Eric.Aligo/UPP>git remote add upstream 
***@***.***:NOAA-EMC/UPP.git
Hera:/scratch1/NCEPDEV/stmp2/Eric.Aligo/UPP>git fetch upstream
Permission denied (publickey).
fatal: Could not read from remote repository.

Please make sure you have the correct access rights
and the repository exists.
Hera:/scratch1/NCEPDEV/stmp2/Eric.Aligo/UPP>git remote -v
origin https://github.com/ericaligo-NOAA/UPP.git (fetch)
origin https://github.com/ericaligo-NOAA/UPP.git (push)
upstream ***@***.***:NOAA-EMC/UPP.git (fetch)
upstream ***@***.***:NOAA-EMC/UPP.git (push)

Is this the correct upstream?

On 12/19/2022 12:38 PM, WenMeng-NOAA wrote:
>
> @ericaligo-NOAA <https://github.com/ericaligo-NOAA> Can you sync your 
> branch with upstream/develop? The procedure can be found at
> https://github.com/NOAA-EMC/UPP/wiki/UPP-Code-Development#update-your-branch-at-your-fork-with-branch-develop-at-authoritative-repository
>
> —
> Reply to this email directly, view it on GitHub 
> <https://github.com/NOAA-EMC/UPP/pull/600#issuecomment-1358012548>, or 
> unsubscribe 
> <https://github.com/notifications/unsubscribe-auth/ALQ75MJ4NOIBN5KXYHHF57DWOCMSVANCNFSM6AAAAAASYSFTNM>.
> You are receiving this because you were mentioned.Message ID: 
> ***@***.***>
>

ericaligo-NOAA commented:
There's a conflict in the flat file. Not sure how to deal with this:

Hera:/scratch1/NCEPDEV/stmp2/Eric.Aligo/UPP>git merge upstream/develop
Auto-merging sorc/ncep_post.fd/VRBLS2D_mod.f
Auto-merging sorc/ncep_post.fd/SURFCE.f
Auto-merging sorc/ncep_post.fd/INITPOST_NETCDF.f
Auto-merging sorc/ncep_post.fd/DEALLOCATE.f
Auto-merging sorc/ncep_post.fd/ALLOCATE_ALL.f
Auto-merging parm/postxconfig-NT-fv3lam_rrfs.txt
*CONFLICT (content): Merge conflict in parm/postxconfig-NT-fv3lam_rrfs.txt*
Removing parm/postxconfig-NT-fv3lam_cmaq.txt
Auto-merging parm/postxconfig-NT-fv3lam.txt
Auto-merging parm/post_avblflds.xml
Auto-merging parm/fv3lam_rrfs.xml
Removing parm/fv3lam_cmaq.xml
Auto-merging parm/fv3lam.xml
Automatic merge failed; fix conflicts and then commit the result.

On 12/19/2022 12:38 PM, WenMeng-NOAA wrote:
>
> @ericaligo-NOAA <https://github.com/ericaligo-NOAA> Can you sync your 
> branch with upstream/develop? The procedure can be found at
> https://github.com/NOAA-EMC/UPP/wiki/UPP-Code-Development#update-your-branch-at-your-fork-with-branch-develop-at-authoritative-repository
>
> —
> Reply to this email directly, view it on GitHub 
> <https://github.com/NOAA-EMC/UPP/pull/600#issuecomment-1358012548>, or 
> unsubscribe 
> <https://github.com/notifications/unsubscribe-auth/ALQ75MJ4NOIBN5KXYHHF57DWOCMSVANCNFSM6AAAAAASYSFTNM>.
> You are receiving this because you were mentioned.Message ID: 
> ***@***.***>
>

ericaligo-NOAA committed:
Remake flat files for gfs f00 and RRFS
Merge remote-tracking branch 'upstream/develop' into winterwx

WenMeng-NOAA commented on parm/postcntrl_gfs_f00_two.xml:
```diff
@@ -907,6 +907,36 @@
        <scale>3.0</scale>
     </param>
 
+    <param>
```
@ericaligo-NOAA Can you revert the changes in this file?

WenMeng-NOAA commented on parm/postxconfig-NT-GFS-F00-TWO.txt:
```diff
@@ -1,6 +1,6 @@
 2
 47
-156
+162
```
@ericaligo-NOAA Can you revert the changes in this file?

ericaligo-NOAA commented:
Just for clarification, I should remove these fields from 
postcntrl_gfs_f00_two.xml?

  910     <param>
  911 <shortname>ACM_GRAUPEL_ON_SURFACE</shortname>
  912        <scale>6.0</scale>
  913     </param>
  914
  915     <param>
  916 <shortname>BUCKET_GRAUPEL_ON_SURFACE</shortname>
  917        <scale>6.0</scale>
  918     </param>
  919
  920     <param>
  921 <shortname>ACM_FRAIN_ON_SURFACE</shortname>
  922        <scale>4.0</scale>
  923     </param>
  924
  925     <param>
  926 <shortname>BUCKET_FRAIN_ON_SURFACE</shortname>
  927        <scale>4.0</scale>
  928     </param>
  929
  930     <param>
  931 <shortname>ACM_SNOWFALL_ON_SURFACE</shortname>
  932        <scale>4.0</scale>
  933     </param>
  934
  935     <param>
  936 <shortname>BUCKET_SNOWFALL_ON_SURFACE</shortname>
  937        <scale>4.0</scale>
  938     </param>

On 12/19/2022 4:39 PM, WenMeng-NOAA wrote:
>
> ***@***.**** commented on this pull request.
>
> ------------------------------------------------------------------------
>
> In parm/postcntrl_gfs_f00_two.xml 
> <https://github.com/NOAA-EMC/UPP/pull/600#discussion_r1052658846>:
>
> > @@ -907,6 +907,36 @@
>          <scale>3.0</scale>
>       </param>
>   
> +    <param>
>
> @ericaligo-NOAA <https://github.com/ericaligo-NOAA> Can you revert the 
> changes in this file?
>
> —
> Reply to this email directly, view it on GitHub 
> <https://github.com/NOAA-EMC/UPP/pull/600#pullrequestreview-1223564163>, 
> or unsubscribe 
> <https://github.com/notifications/unsubscribe-auth/ALQ75MJJZG3TOEDHUU2SG4TWODIX7ANCNFSM6AAAAAASYSFTNM>.
> You are receiving this because you were mentioned.Message ID: 
> ***@***.***>
>

WenMeng-NOAA commented:
> Just for clarification, I should remove these fields from postcntrl_gfs_f00_two.xml?  910     <param>  911 <shortname>ACM_GRAUPEL_ON_SURFACE</shortname>  912        <scale>6.0</scale>  913     </param>  914  915     <param>  916 <shortname>BUCKET_GRAUPEL_ON_SURFACE</shortname>  917        <scale>6.0</scale>  918     </param>  919  920     <param>  921 <shortname>ACM_FRAIN_ON_SURFACE</shortname>  922        <scale>4.0</scale>  923     </param>  924  925     <param>  926 <shortname>BUCKET_FRAIN_ON_SURFACE</shortname>  927        <scale>4.0</scale>  928     </param>  929  930     <param>  931 <shortname>ACM_SNOWFALL_ON_SURFACE</shortname>  932        <scale>4.0</scale>  933     </param>  934  935     <param>  936 <shortname>BUCKET_SNOWFALL_ON_SURFACE</shortname>  937        <scale>4.0</scale>  938     </param>
> […](#)
> On 12/19/2022 4:39 PM, WenMeng-NOAA wrote: ***@***.**** commented on this pull request. ------------------------------------------------------------------------ In parm/postcntrl_gfs_f00_two.xml <[#600 (comment)](https://github.com/NOAA-EMC/UPP/pull/600#discussion_r1052658846)>: > @@ -907,6 +907,36 @@ <scale>3.0</scale> </param> + <param> @ericaligo-NOAA <https://github.com/ericaligo-NOAA> Can you revert the changes in this file? — Reply to this email directly, view it on GitHub <[#600 (review)](https://github.com/NOAA-EMC/UPP/pull/600#pullrequestreview-1223564163)>, or unsubscribe <https://github.com/notifications/unsubscribe-auth/ALQ75MJJZG3TOEDHUU2SG4TWODIX7ANCNFSM6AAAAAASYSFTNM>. You are receiving this because you were mentioned.Message ID: ***@***.***>

Yes, this control file is used for gfs inline post.

ericaligo-NOAA committed:
remove winter weather diags from gfs f00 xml and txt files

ericaligo-NOAA commented:
Done.

On 12/19/2022 8:12 PM, WenMeng-NOAA wrote:
>
>     Just for clarification, I should remove these fields from
>     postcntrl_gfs_f00_two.xml?  910      911 ACM_GRAUPEL_ON_SURFACE
>      912        6.0  913      914  915  916 BUCKET_GRAUPEL_ON_SURFACE
>      917        6.0  918      919  920      921 ACM_FRAIN_ON_SURFACE
>      922        4.0  923  924  925      926 BUCKET_FRAIN_ON_SURFACE
>      927        4.0  928      929  930      931
>     ACM_SNOWFALL_ON_SURFACE  932        4.0  933      934  935    
>      936 BUCKET_SNOWFALL_ON_SURFACE  937        4.0  938
>     … <#>
>     On 12/19/2022 4:39 PM, WenMeng-NOAA wrote: /*@*/.**** commented on
>     this pull request.
>     ------------------------------------------------------------------------
>     In parm/postcntrl_gfs_f00_two.xml <#600 (comment)
>     <https://github.com/NOAA-EMC/UPP/pull/600#discussion_r1052658846>>:
>     > @@ -907,6 +907,36 @@ 3.0 + @ericaligo-NOAA
>     <https://github.com/ericaligo-NOAA>
>     https://github.com/ericaligo-NOAA Can you revert the changes in
>     this file? — Reply to this email directly, view it on GitHub <#600
>     (review)
>     <https://github.com/NOAA-EMC/UPP/pull/600#pullrequestreview-1223564163>>,
>     or unsubscribe
>     https://github.com/notifications/unsubscribe-auth/ALQ75MJJZG3TOEDHUU2SG4TWODIX7ANCNFSM6AAAAAASYSFTNM.
>     You are receiving this because you were mentioned.Message ID:
>     /*@*/.***>
>
> Yes, this control file is used for gfs inline post.
>
> —
> Reply to this email directly, view it on GitHub 
> <https://github.com/NOAA-EMC/UPP/pull/600#issuecomment-1358701098>, or 
> unsubscribe 
> <https://github.com/notifications/unsubscribe-auth/ALQ75MPGQ2QWEZKTVAD2OALWOEBWBANCNFSM6AAAAAASYSFTNM>.
> You are receiving this because you were mentioned.Message ID: 
> ***@***.***>
>

WenMeng-NOAA commented on sorc/ncep_post.fd/INITPOST_NETCDF.f:
```diff
@@ -1620,6 +1621,35 @@ SUBROUTINE INITPOST_NETCDF(ncid2d,ncid3d)
       spval,VarName,SFCEXC)
 
+! accumulated snowfall
+      VarName='tsnowp'
```
@EricJames-NOAA You will submit a PR to output some GSL precip type variables for RRFS.  I would like to know if there are any conflicts since 3 GSL variables (acgraup, graup_bucket, acfrain) were assigned to read new winter diag fields from model.

ericaligo-NOAA commented:
Hi, there was a meeting a few weeks ago with GSL discussing the fields 
that would be output. Here's the document I provided and slide 2 is what 
was agreed upon.

On 12/20/2022 11:48 AM, WenMeng-NOAA wrote:
>
> ***@***.**** commented on this pull request.
>
> ------------------------------------------------------------------------
>
> In sorc/ncep_post.fd/INITPOST_NETCDF.f 
> <https://github.com/NOAA-EMC/UPP/pull/600#discussion_r1053532078>:
>
> > @@ -1620,6 +1621,35 @@ SUBROUTINE INITPOST_NETCDF(ncid2d,ncid3d)
>         call read_netcdf_2d_para(ncid2d,ista,ista_2l,iend,iend_2u,jsta,jsta_2l,jend,jend_2u, &
>         spval,VarName,SFCEXC)
>   
> +! accumulated snowfall
> +      VarName='tsnowp'
>
> @EricJames-NOAA <https://github.com/EricJames-NOAA> You will submit a 
> PR to output some GSL precip type variables for RRFS. I would like to 
> know if there are any conflicts since 3 GSL variables (acgraup, 
> graup_bucket, acfrain) were assigned to read new winter diag fields 
> from model.
>
> —
> Reply to this email directly, view it on GitHub 
> <https://github.com/NOAA-EMC/UPP/pull/600#pullrequestreview-1224817968>, 
> or unsubscribe 
> <https://github.com/notifications/unsubscribe-auth/ALQ75MMUOAXT5QA6NXHCJEDWOHPNTANCNFSM6AAAAAASYSFTNM>.
> You are receiving this because you were mentioned.Message ID: 
> ***@***.***>
>

ericaligo-NOAA committed:
Add one line to both acfrain and acgraup sections to ensure the correct time stamp in grib2 file for accumulated fields.

WenMeng-NOAA commented on sorc/ncep_post.fd/INITPOST_NETCDF.f:
```diff
@@ -53,7 +53,8 @@ SUBROUTINE INITPOST_NETCDF(ncid2d,ncid3d)
               uz0, vz0, ptop, htop, pbot, hbot, ptopl, pbotl, ttopl, ptopm, pbotm, ttopm,       &
               ptoph, pboth, pblcfr, ttoph, runoff, tecan, tetran, tedir, twa, maxtshltr,        &
-              mintshltr, maxrhshltr, fdnsst,            &
+              mintshltr, maxrhshltr, fdnsst, acgraup, graup_bucket, acfrain, frzrn_bucket,      &
```
@ericaligo-NOAA Can you add a change log with your name at the start session of this subroutine? e.g. !> 2022-11-16 | Eric James    | ...

WenMeng-NOAA commented on sorc/ncep_post.fd/SURFCE.f:
```diff
@@ -93,6 +93,7 @@ SUBROUTINE SURFCE
                          acond,maxqshltr,minqshltr,avgpotevp,AVGPREC_CONT,    &
```
@ericaligo-NOAA Can you add a change log in this subroutine?

ericaligo-NOAA committed:
Add change log to SURFCE.f and INITPOST_NETCDF.f

ericaligo-NOAA commented:
Added change log to both INITPOST and SURFCE.f

On 12/23/2022 10:51 AM, WenMeng-NOAA wrote:
>
> ***@***.**** commented on this pull request.
>
> ------------------------------------------------------------------------
>
> In sorc/ncep_post.fd/SURFCE.f 
> <https://github.com/NOAA-EMC/UPP/pull/600#discussion_r1056437101>:
>
> > @@ -93,6 +93,7 @@ SUBROUTINE SURFCE
>                            acond,maxqshltr,minqshltr,avgpotevp,AVGPREC_CONT,    &
>
> @ericaligo-NOAA <https://github.com/ericaligo-NOAA> Can you add a 
> change log in this subroutine?
>
> —
> Reply to this email directly, view it on GitHub 
> <https://github.com/NOAA-EMC/UPP/pull/600#pullrequestreview-1229120544>, 
> or unsubscribe 
> <https://github.com/notifications/unsubscribe-auth/ALQ75MLS27PEJSBJDB3WRCLWOXC73ANCNFSM6AAAAAASYSFTNM>.
> You are receiving this because you were mentioned.Message ID: 
> ***@***.***>
>

WenMeng-NOAA commented:
The UPP RT tests were completed on WCOSS2, Hera and Orion. The updates of baselines for GFS and RRFS are needed with this PR.
